test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,615 @@ describe("BaseAxis", () => {
+            });
+            svg.remove();
+          });
+        });
+
+        describe("first row offset position", () => {
+          it("places the first row right at the beginning of the margin for bottom orientation", () => {
+            let axis = new Plottable.Axis(mockScale, "bottom");
+            axis.annotatedTicks(annotatedTicks);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let firstAnnotationRect = axis.content().select(".annotation-rect");
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "y"),
+              axis.height() - axis.margin(), "rectangle positioned correctly");
+            svg.remove();
+          });
+
+          it("places the first row right at the beginning of the margin for top orientation", () => {
+            let axis = new Plottable.Axis(mockScale, "top");
+            axis.annotatedTicks(annotatedTicks);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let firstAnnotationRect = axis.content().select(".annotation-rect");
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "y") + TestMethods.numAttr(firstAnnotationRect, "height"),
+              axis.margin(), "rectangle positioned correctly");
+            svg.remove();
+          });
+
+          it("places the first row right at the beginning of the margin for left orientation", () => {
+            let axis = new Plottable.Axis(mockScale, "left");
+            axis.annotatedTicks(annotatedTicks);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let firstAnnotationRect = axis.content().select(".annotation-rect");
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "x") + TestMethods.numAttr(firstAnnotationRect, "width"),
+              axis.margin(), "rectangle positioned correctly");
+            svg.remove();
+          });
+
+          it("places the first row right at the beginning of the margin for right orientation", () => {
+            let axis = new Plottable.Axis(mockScale, "right");
+            axis.annotatedTicks(annotatedTicks);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let firstAnnotationRect = axis.content().select(".annotation-rect");
+            assert.strictEqual(TestMethods.numAttr(firstAnnotationRect, "x"),
+              axis.width() - axis.margin(), "rectangle positioned correctly");
+            svg.remove();
+          });
+        });
+
+        describe("additional row offset position", () => {
+          it("offsets additional rows by the placement of colliding rectangles for bottom orientation", () => {
```

Comment:
It's hard for me to understand what this test is doing based on the description. Is this correct:

`"moves rectangles that would overlap to different rows (orientation: bottom)"`